fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVService.java

```diff
@@ -207,12 +209,19 @@ public void processEvent(Event event) throws EventException {
         }
     }
 
-    private boolean canRefresh(MTMV mtmv, TableIf table)   {
+    private boolean canRefresh(MTMV mtmv, TableIf table) {
         if (mtmv.getExcludedTriggerTables().contains(table.getName())) {
             LOG.info("skip refresh mtmv: {}, because exclude trigger table: {}",
                     mtmv.getName(), table.getName());
             return false;
         }
+        // replace/alter base table,not change MTMVRelation, only change MTMV to schema_change,
+        // Therefore, it may trigger incorrect materialized view refresh
+        if (mtmv.getStatus().getState().equals(MTMVState.SCHEMA_CHANGE)) {
+            LOG.info("skip refresh mtmv: {}, because state is SCHEMA_CHANGE, trigger table: {}",
+                    mtmv.getName(), table.getName());
+            return false;
```

Comment:
why we check here, i think it is better to check it when we run refresh task?

Reply:
This is to check if a task should be generated, as it cannot be checked while the task is running